Sources/IndexStoreDB/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+add_library(IndexStoreDB
+  IndexStoreDB.swift
+  SymbolLocation.swift
+  SymbolOccurrence.swift
+  SymbolRole.swift
+  Symbol.swift)
+set_target_properties(IndexStoreDB PROPERTIES
+  Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift
+  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_BINARY_DIR}/swift)
+target_link_libraries(IndexStoreDB PRIVATE
+  Foundation
+  CIndexStoreDB)
+
+get_swift_host_arch(swift_arch)
+install(TARGETS IndexStoreDB
```

Comment:
Why are we installing indexstore-db independently? In SourceKit-LSP it's always statically linked.  The install logic seems like the most complex part of this whole PR.

Reply:
Oh, I didn't realize that IndexStoreDB was supposed to be a static library based on `Package.swift`.  One wrinkle is that Windows does not support static swift libraries currently, so this needs to be a shared object.  Yes, the install logic is the bulk of the logic :)